src/core/thread/link_metrics.cpp

```diff
@@ -287,7 +288,7 @@ Error LinkMetrics::AppendReport(Message &aMessage, const Message &aRequestMessag
 
     if (queryId == kQueryIdSingleProbe)
     {
-        values.mPduCountValue = aRequestMessage.GetPsduCount();
+        values.mPduCountValue = ot::Encoding::BigEndian::HostSwap32(aRequestMessage.GetPsduCount());
```

Comment:
nit: should we leverage `using` to avoid needing fully-qualified name?

For example:
https://github.com/openthread/openthread/blob/dc7676a032ccd5610fbb0141c0c41c49cbb76710/src/core/net/ip6_address.cpp#L46